server/remote/gcal/get_default_calendar_view.go

```diff
@@ -0,0 +1,224 @@
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/remote"
+)
+
+const (
+	RemoteEventBusy = "busy"
+	RemoteEventFree = "free"
+
+	GoogleEventBusy = "opaque"
+	GoogleEventFree = "transparent"
+
+	ResponseYes   = "accepted"
+	ResponseMaybe = "tentativelyAccepted"
+	ResponseNo    = "declined"
+	ResponseNone  = "notResponded"
+
+	GoogleResponseStatusYes   = "accepted"
+	GoogleResponseStatusMaybe = "tentative"
+	GoogleResponseStatusNo    = "declined"
+	GoogleResponseStatusNone  = "needsAction"
+)
+
+var responseStatusConversion = map[string]string{
+	GoogleResponseStatusYes:   ResponseYes,
+	GoogleResponseStatusMaybe: ResponseMaybe,
+	GoogleResponseStatusNo:    ResponseNo,
+	GoogleResponseStatusNone:  ResponseNone,
+}
+
+func (c *client) GetDefaultCalendarView(remoteUserID string, start, end time.Time) ([]*remote.Event, error) {
+	service, err := calendar.New(c.httpClient)
+	if err != nil {
+		return nil, errors.Wrap(err, "gcal GetDefaultCalendarView, error creating service")
+	}
+
+	req := service.Events.List("primary")
+	req.MaxResults(20)
+	req.TimeMin(start.Format(time.RFC3339))
+	req.TimeMax(end.Format(time.RFC3339))
+	req.SingleEvents(true)
+	req.OrderBy("startTime")
+
+	events, err := req.Do()
+	if err != nil {
+		return nil, errors.Wrap(err, "gcal GetDefaultCalendarView, error performing request")
+	}
+
+	result := []*remote.Event{}
+	if len(events.Items) == 0 {
+		return result, nil
+	}
+
+	for _, event := range events.Items {
+		if event.ICalUID != "" {
+			result = append(result, convertGCalEventToRemoteEvent(event))
+		}
+	}
+
+	return result, nil
+}
+
+func convertGCalEventToRemoteEvent(event *calendar.Event) *remote.Event {
+	showAs := RemoteEventBusy
+	if event.Transparency == GoogleEventFree {
+		showAs = RemoteEventFree
+	}
+
+	start := remote.NewGoogleDateTime(event.Start)
+	end := remote.NewGoogleDateTime(event.End)
+
+	location := &remote.Location{
+		DisplayName: event.Location,
+	}
+
+	organizer := &remote.Attendee{
+		EmailAddress: &remote.EmailAddress{
+			Name:    event.Organizer.Email,
+			Address: event.Organizer.Email,
+		},
+	}
+
+	var responseStatus *remote.EventResponseStatus
+	responseRequested := false
+	isOrganizer := false
+
+	attendees := []*remote.Attendee{}
+	for _, attendee := range event.Attendees {
+		attendees = append(attendees, &remote.Attendee{
+			Status: &remote.EventResponseStatus{
+				Response: attendee.ResponseStatus,
+			},
+			EmailAddress: &remote.EmailAddress{
+				Name:    attendee.Email,
+				Address: attendee.Email,
+			},
+		})
+
+		if attendee.Self {
+			if attendee.ResponseStatus == GoogleResponseStatusNone {
+				responseRequested = true
+			}
+
+			response := responseStatusConversion[attendee.ResponseStatus]
+			responseStatus = &remote.EventResponseStatus{
+				Response: response,
+			}
+
+			isOrganizer = attendee.Organizer
+		}
+	}
+
+	isAllDay := len(event.Start.Date) > 0 // if Date field is present, it is all-day. as opposed to DateTime field
+
+	return &remote.Event{
```

Comment:
Does it make sense to keep calling that package remote? The way I read it, the remote event is the gcal/mscal event. And I want to convert that to a "mattermost plugin readable event". Right?

Reply:
@larkox Are you talking about `remote.Event` specifically (and possibly other contents of `remote` package), and not the `remote/gcal` or `remote/msgraph` packages? I think it makes sense to have things specifically related to remote communication in those packages. But common data models within `remote` can go elsewhere.

As far as things like `remote.Event`, it is basically based off of Microsoft's data model, so it was originally seen as a "remote" event, but now it is indeed just a "mattermost plugin readable event" as you mentioned. It (and other structs like it) can go into a more shareable package like a new package named `model`.